pkg/scheduler/framework/framework_test.go

```diff
@@ -0,0 +1,358 @@
+			binding: binding,
+			policy: &fleetv1beta1.ClusterPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: policyName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							Kind: utils.CRPV1Beta1GVK.Kind,
+							Name: CRPName,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "no owner reference in policy",
+			policy: &fleetv1beta1.ClusterPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: policyName,
+				},
+			},
+			expectToFail: true,
+		},
+		{
+			name:    "no matching bindings",
+			binding: binding,
+			policy: &fleetv1beta1.ClusterPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: policyName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							Kind: utils.CRPV1Beta1GVK.Kind,
+							Name: altCRPName,
+						},
+					},
+				},
+			},
+			expectToFindNoBindings: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeClientBuilder := fake.NewClientBuilder().WithScheme(scheme.Scheme)
+			if tc.binding != nil {
+				fakeClientBuilder.WithObjects(tc.binding)
+			}
+			fakeClient := fakeClientBuilder.Build()
+			// Construct framework manually instead of using NewFramework() to avoid mocking the controller manager.
+			f := &framework{
+				uncachedReader: fakeClient,
+			}
+
+			ctx := context.Background()
+			bindings, err := f.collectBindings(ctx, tc.policy)
+			if tc.expectToFail {
+				if err == nil {
+					t.Fatalf("collectBindings() = %v, %v, want failed to collect bindings error", bindings, err)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Fatalf("collectBindings() = %v, %v, want %v, no error", bindings, err, binding)
+			}
+			wantBindings := []fleetv1beta1.ClusterResourceBinding{}
+			if !tc.expectToFindNoBindings {
+				wantBindings = append(wantBindings, *binding)
+			}
+			if !cmp.Equal(bindings, wantBindings) {
+				t.Fatalf("collectBindings() = %v, %v, want %v, no error", bindings, err, wantBindings)
+			}
+		})
+	}
+}
+
+// TestClassifyBindings tests the classifyBindings function.
+func TestClassifyBindings(t *testing.T) {
+	timestamp := metav1.Now()
+	activeBinding := &fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "active-binding",
+		},
+	}
+	deletedBindingWithDispatcherFinalizer := &fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "deleted-binding-with-dispatcher-finalizer",
+			DeletionTimestamp: &timestamp,
+			Finalizers: []string{
+				utils.DispatcherFinalizer,
+			},
+		},
+	}
+	deletedBindingWithoutDispatcherFinalizer := &fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "deleted-binding-without-dispatcher-finalizer",
+			DeletionTimestamp: &timestamp,
+		},
+	}
+
+	active, deletedWith, deletedWithout := classifyBindings([]fleetv1beta1.ClusterResourceBinding{*activeBinding, *deletedBindingWithDispatcherFinalizer, *deletedBindingWithoutDispatcherFinalizer})
```

Comment:
add a test for the empty binding list?

Reply:
As mentioned the in the description, per our earlier agreement, at this moment PRs will be featuring limited test cases. More will be added in future PRs, when the development settles a little bit.